azext_edge/edge/util/az_client.py

```diff
@@ -183,6 +184,23 @@ def get_authz_client(subscription_id: str, **kwargs) -> "AuthorizationManagement
     )
 
 
+def get_keyvault_client(subscription_id: str, keyvault_name: str, **kwargs) -> "SecretClient":
+    from azure.keyvault.secrets import SecretClient
+
+    client = SecretClient(
+        credential=AZURE_CLI_CREDENTIAL,
+        subscription_id=subscription_id,
+        user_agent_policy=UserAgentPolicy(user_agent=USER_AGENT),
+        vault_url=f"https://{keyvault_name}.vault.azure.net",
```

Comment:
will this always be the vault url? double check if there is a "canary" or some other url format

Reply:
added a todo, current plan is to only focus on azure public cloud